### ISSUE TITLE
catboost 1.0.3 use gpu is slower than cpu

### ISSUE THREAD

**Issue by K1A2** [catboost/catboost#1921]
Problem: catboost 1.0.3 use gpu is slower than cpu
catboost version: 1.0.3
Operating System: Windows 10 pro
CPU: AMD Ryzen 5600X
GPU: GTX 1650 4gb, CUDA 11.5

If i training CatBoostClassifier with gpu, it takes more than a day. But with cpu, it's just a few hours faster.
I think gpu should be much faster than cpu. But now cpu is faster. Why is this happening?
Is this a bug?

**Comment by LyzhinIvan** [comment]
Hi, @K1A2!

Please tell more details about your task. What are training parameters? What is the size of your dataset? Which features does it have - numeric, categorical, texts, embeddings?

**Comment by K1A2** [comment]
type: DataFrame
shape: (26457, 34)
categorical feature counts: 11
numerical feature counts: 16
parameters:
param = {
        'objective': 'MultiClass',
        'depth': 10,
        'learning_rate': 0.06576219655285793,
        'grow_policy': 'Lossguide',
        'bootstrap_type': 'Bernoulli',
        'l2_leaf_reg': 5,
        'subsample': 0.692762589836913,
        'max_leaves': 56,
        'iterations': 40000,
    }
Label encoding all categorical features
GaussianRankScaling all numerical features

**Comment by LyzhinIvan** [comment]
> categorical feature counts: 11
   Label encoding all categorical features

Do I understand correctly - you don't specify `cat_features` option and pass all features as numeric actually? 


> 'objective': 'MultiClass'

How many classes?


> shape: (26457, 34)

Actually, this is pretty small dataset to get significant difference between CPU and GPU.

**Comment by K1A2** [comment]
I use Pool(cat_features=[categorical columns]) and 
the classes count is 3.

**Comment by K1A2** [comment]
If i run same code in ubuntu, this problem doesn't happen.

**Comment by LyzhinIvan** [comment]
> If i run same code in ubuntu, this problem doesn't happen.

With the same GPU? What is the driver version on Windows and Ubuntu?
How much time does training take on Windows-CPU, Windows-GPU, Ubuntu-CPU, Ubuntu-GPU?

**Comment by K1A2** [comment]
yes, same gpu.
ubuntu version is 20.04.

Driver version
- Windows: 472.47
- Ubuntu: 460.91.03

times (one epoch)
- Windows-CPU: 4h
- Windows-GPU: 5h
- Ubuntu-GPU: 2h~3h

**Comment by LyzhinIvan** [comment]
What is Ubuntu-CPU time?

> (one epoch)

one iteration?

**Comment by K1A2** [comment]
yes, one iteration

**Comment by Iqigai** [comment]
I have the same problem on Windows 10. It takes much longer with catboost 1.0.3 to train on GPU (1m19s) than on CPU (9s). The task is quite simple, consisting in binary classification of a 10,000 rows by 13 features data set. Here is a screenshot.

![image](https://user-images.githubusercontent.com/80190909/143081119-d954d620-5899-4cfc-86b6-085d400fd2ba.png)

**Comment by sanmaoyang** [comment]
any solution??

**Comment by Evgueni-Petrov-aka-espetrov** [comment]
dear @sanmaoyang 
> any solution??

datasets of 10-100K samples are too small for gpu to outperform cpu in training
this is because of kernel launch and cuda memcpy overhead, which over-weigh by far kernel execution on gpu

**Comment by sanmaoyang** [comment]
Thank you for your suggestion, you are right, we should use GPU accelaration when we have a huge dataset(more than 1 billion). If you have a relative small dataset but you want a high iteration, you had better use CPU with thread_count = -1.
